### PR TITLE
#385 csp connect-src

### PR DESCRIPTION
**Pull request by Mizzick**
#385

### CODE REVIEW

File: Extension/browser/webkit/prefs.js [hunk 1]
```diff
@@ -67,6 +67,18 @@ var Prefs = exports.Prefs = {
 				};
 
 				Prefs.safariVersion = parseSafariVersion();
+			} else if (browser == "Chrome") {
+				var parseChromeVersion = function() {
```
Comment (ameshkov) [inline Extension/browser/webkit/prefs.js]: 0?

File: Extension/browser/webkit/prefs.js [hunk 2]
```diff
@@ -67,6 +67,18 @@ var Prefs = exports.Prefs = {
...
+					var i = userAgent.indexOf("Chrome/");
+					if (i === 0) {
```
Comment (ameshkov) [inline Extension/browser/webkit/prefs.js]: 0?

File: Extension/browser/webkit/webrequest.js [hunk 3]
```diff
@@ -114,9 +114,11 @@ function onHeadersReceived(requestDetails) {
          WS connections are detected as "other"  by ABP
          EasyList already contains some rules for WS connections with $other modifier
          */
+        var chromeVersion = Prefs.chromeVersion;
+        var blockConnectSrc = chromeVersion && chromeVersion.substring(0, chromeVersion.indexOf('.')) > 51;
```
Comment (ameshkov) [inline Extension/browser/webkit/webrequest.js]: I suppose there should be a comment linking this issue

File: Extension/browser/webkit/webrequest.js [hunk 4]
```diff
@@ -114,9 +114,11 @@ function onHeadersReceived(requestDetails) {
          WS connections are detected as "other"  by ABP
          EasyList already contains some rules for WS connections with $other modifier
          */
+        var chromeVersion = Prefs.chromeVersion;
+        var blockConnectSrc = chromeVersion && chromeVersion.substring(0, chromeVersion.indexOf('.')) > 51;
         var websocketCheckUrl = "ws://adguardwebsocket.check/";
         if (webRequestService.checkWebSocketRequest(tab, websocketCheckUrl, referrerUrl)) {
-            if (CspUtils.blockWebSockets(responseHeaders)) {
+            if (CspUtils.blockWebSockets(responseHeaders, !blockConnectSrc)) {
```
Comment (ameshkov) [inline Extension/browser/webkit/webrequest.js]: Also I think you should check chrome version right here before calling the method.

**Comment by Mizzick** [comment]
Ok

File: Extension/browser/webkit/prefs.js [hunk 5]
```diff
@@ -67,6 +67,18 @@ var Prefs = exports.Prefs = {
...
 				Prefs.safariVersion = parseSafariVersion();
+			} else if (browser == "Chrome") {
+				var parseChromeVersion = function() {
+					var i = userAgent.indexOf("Chrome/");
+					if (i < 0) {
+						return "";
+					}
+
+					var end = userAgent.indexOf(" ", i);
+					return userAgent.substring(i + 7, end > 0 ? end : userAgent.length);
```
Comment (ameshkov) [inline Extension/browser/webkit/prefs.js]: parseInt maybe?

File: Extension/browser/webkit/webrequest.js [hunk 6]
```diff
@@ -116,7 +116,12 @@ function onHeadersReceived(requestDetails) {
          */
         var websocketCheckUrl = "ws://adguardwebsocket.check/";
         if (webRequestService.checkWebSocketRequest(tab, websocketCheckUrl, referrerUrl)) {
-            if (CspUtils.blockWebSockets(responseHeaders)) {
+
+            //We don't change connect-src directives for older chromes
+            //https://github.com/AdguardTeam/AdguardBrowserExtension/issues/385
+            var blockConnectSrc = parseInt(Prefs.chromeVersion) > 51;
```
Comment (ameshkov) [inline Extension/browser/webkit/webrequest.js]: Argh, why here? Why not in prefs? The same question for safariversion

**Pull request merged**